### ISSUE TITLE
[BUG] - Cli prevents spending from script with inline datum at same utxo

### ISSUE THREAD

**Issue by james-iohk** [input-output-hk/cardano-node#3917]
**Internal/External**
*Internal*   if an IOHK staff member.

**Area**
*Plutus*    Related to Plutus Scripts (Alonzo).

**Summary**
The cli enforces including datum with a spending script input. This combined with the limitation where inline datum cannot also exist in the witness set ([see here](https://github.com/input-output-hk/cardano-ledger/blob/e366ea90c6757a3b526efb7f8112987b318d7b5e/libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/BabbageFeatures.hs#L679)) means it is not possible to consume/reference an output with inline datum to satisfy the spending script. Including datum for spending script should be optional.

**Steps to reproduce**
1. With on output with inline datum at a script address that satisfies the script
2. Attempt to spend the script output with the cli

**Actual result**
When using `--tx-in-script-file` you must provide `--tx-in-datum-*`. Doing so will result in the following ledger error:
```
ShelleyTxValidationError ShelleyBasedEraBabbage (ApplyTxError [UtxowFailure (FromAlonzoUtxowFail (NonOutputSupplimentaryDatums (fromList [SafeHash "9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b"]) (fromList [])))])
```

**Expected behavior**
It is possible to use `--tx-in-script-file` and omit `--tx-in-datum-*` so that only the inline datum is used for script execution, or perhaps you need to also reference the same input (without needing to attach datum).

**System info (please complete the following information):**
- Node version d0b0fa10ac83dc1d0d40aaf294cfded1b455a0a4 [vasil-testnet-v1](https://github.com/input-output-hk/cardano-node/releases/tag/vasil-testnet-v1)

**Comment by james-iohk** [comment]
**We should also fix this for reference inputs**
It'll make `--reference-tx-in-datum-*` optional.

**Comment by james-iohk** [comment]
Resolved by #4006
